File: solution/1863. Sum of All Subset XOR Totals/1863_02.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        const int n = nums.size();
        int res = 0;
        for (int state = 0; state < (1 << n); ++state) {
            int x = 0;
            for (int i = 0; i < n; ++i) {
                if ((state >> i) & 1)
                    x ^= nums[i];
            }
            res += x;
        }
        return res;
    }
};

File: solution/1863. Sum of All Subset XOR Totals/1863_02.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        for state in range(1 << n):
            x = 0
            for i in range(n):
                if state & 1:
                    x ^= nums[i]
                state >>= 1
            res += x
        return res
    
